chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -516,6 +512,37 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
         Ok(())
     }
 
+    fn fetch_block_undo<'b>(
+        db_tx: &'a S,
+        utxo_block_undo: &'b mut BTreeMap<Id<Block>, BlockUndoEntry>,
```

Comment:
borrows checker is complaining that the utxo_cache field is borrowed here (though it is not used) and in the next line after this function call

Reply:
The idea was that I return a ref to an entry in the map to avoid copying or additional lookup. I don't see a design issue here, besides that such style with refs is avoided in rust. 
If you believe that I overthink the problem and I should just return by value I can rewrite this.

Reply:
The better way is to pop the entries you're disconnecting one by one by value since you're disconnecting them one by one. Logically, if you're gonna use TxUndo entries and put them back into the utxo set, then taking them by value only makes sense. Why take their reference and then use the self reference to do clone inside it? This was my guess of what you were doing.

With a little digging, I noticed that inside the disconnect_transaction function, you're cloning the utxo! The utxo from TxUndo! This is exactly the design issue rust is trying to protect you from. If you're going to do a clone eventually, maybe it makes more sense to pass things by value, or at least do the cloning before using multiple mutable and immutable references together. That way, you don't have complex references all around the place.